libs/providers/flagd-web/src/lib/local-storage-flagd-cache.ts

```diff
@@ -0,0 +1,98 @@
+import { EvaluationContext, FlagValue, ResolutionDetails } from '@openfeature/js-sdk';
+import { Md5 } from 'ts-md5';
+import { FlagdCache } from './flagd-cache';
+
+type CacheItem<T = FlagValue> = {
+  expiry: number;
+  details: ResolutionDetails<T>;
+  variant: string;
+};
+
+const PREXFIX = '@OpenFeature/flagd-';
+const SEPARATOR = '--';
+const DEFAULT_OPTIONS = {
+  expiryCheckInterval: 60000,
+  itemTtl: 300,
+};
+
+export interface Options {
+  expiryCheckInterval: number;
+  itemTtl: number;
+}
+
+function getOptions(options: Partial<Options> = {}) {
+  return {
+    ...DEFAULT_OPTIONS,
+    ...options,
+  };
+}
+
+export class LocalStorageFlagdCache implements FlagdCache {
+  private _ttl: number | undefined;
+
+  constructor(options?: Partial<Options>) {
+    const defaultedOptions = getOptions(options);
+    this._ttl = defaultedOptions.itemTtl === 0 ? undefined : defaultedOptions.itemTtl;
+    setInterval(() => {
+      const currentTimestamp = new Date().getTime();
+
+      Object.entries(localStorage).forEach((each) => {
+        if (each[0].startsWith(PREXFIX)) {
+          const item: CacheItem = JSON.parse(each[1]);
+          if (currentTimestamp > item.expiry) {
+            localStorage.removeItem(each[0]);
+          }
+        }
+      });
+    }, defaultedOptions.expiryCheckInterval);
+  }
+
+  private buildKey(flagKey: string, context: EvaluationContext) {
+    return `${PREXFIX}${flagKey}${SEPARATOR}${Md5.hashStr(JSON.stringify(context))}`;
+  }
+
+  set<T extends FlagValue>(key: string, context: EvaluationContext, details: ResolutionDetails<T>): Promise<void> {
+    const cacheKey = this.buildKey(key, context);
+    localStorage.setItem(
+      cacheKey,
+      JSON.stringify({
+        expiry: this._ttl ? new Date(new Date().getTime() + this._ttl).getTime() : undefined,
+        details: { ...details, reason: 'CACHED' }
```

Comment:
Cached values are always return with reason `CACHED`.

Reply:
Should we include cached in the example section of the spec?

Reply:
Probably if we merge the SDK changes associated with this.